pkg/network/server.go

```diff
@@ -870,7 +889,7 @@ func (s *Server) handleVersionCmd(p Peer, version *payload.Version) error {
 
 // handleBlockCmd processes the block received from its peer.
 func (s *Server) handleBlockCmd(p Peer, block *block.Block) error {
-	if s.syncHeaderFetcher.IsActive() || s.syncBlockFetcher.IsActive() || s.blockFetcher.IsActive() {
+	if s.syncHeaderFetcher.IsActive() || s.syncBlockFetcher.IsActive() || s.blockFetcher.IsActive() || s.syncStateFetcher.IsActive() || s.stateSync.IsActive() && s.config.NeoFSStateSyncExtensions {
```

Comment:
`|| s.syncStateFetcher.IsActive()` should be enough. Why `s.stateSync.IsActive() && s.config.NeoFSStateSyncExtensions `?